src/NumberIncrement/index.js

```diff
@@ -0,0 +1,174 @@
+import React, { Component } from "react";
+import withStyles from "react-jss";
+import classNames from "classnames";
+import Remove from "elevate-ui-icons/Remove";
+import Add from "elevate-ui-icons/Add";
+
+import Scaffold from "../Scaffold";
+
+type Props = {
+  classes: Object,
+  /**
+   * Accepts any classNames to be passed down to the component.
+   */
+  className: string,
+  field: Object,
+  form: Object,
+  id: string,
+  /**
+   * Text input to be used as the label for inside the component.
+   */
+  label: string,
+  theme: Object,
+  /**
+   * Use the scaffold for the input or if the input should just be the raw input.
+   */
+  withScaffold: boolean,
+};
+
+/**
+ * Number input that can be adjusted incrementally
+ */
+class NumberIncrement extends Component<Props> {
+  handleValueChange = (direction) => {
+    direction === "increase" ? this.input.stepUp() : this.input.stepDown();
+    return this.props.form.setFieldValue(
+      this.props.field.name,
+      this.input.value
+    );
+  };
+
+  render() {
+    const {
+      classes,
+      className,
+      field,
+      form: { touched, errors },
+      id,
+      label,
+      theme,
+      withScaffold = true,
+      ...rest
+    } = this.props;
+
+    if (withScaffold) {
+      return (
+        <Scaffold
+          id={id}
+          label={label}
+          error={touched[field.name] && errors[field.name]}
+        >
+          <div className={classNames(classes.root, className)}>
```

Comment:
Let's declare this entire root div + buttons/input as a `const`, then do our `withScaffold` check to decide whether we need to wrap our new constant in a Scaffold component or just return it.